documentation/design-docs/ipc-protocol.md

```diff
@@ -0,0 +1,183 @@
+# Event Pipe
+
+## Overview
+
+> TODO: Write an overview of what this does
+
+## TODOs
+* formalize struct names
+* formalize error codes (Use HRESULT values?)
+
+## Inter-Process Communication Specification
+
+### Current
+
+```c++
+enum class DiagnosticMessageType : uint32_t
+{
+    // EventPipe
+    StartEventPipeTracing = 1024, // To file
+    StopEventPipeTracing,
+    CollectEventPipeTracing, // To IPC
+};
+
+struct MessageHeader
+{
+    DiagnosticMessageType RequestType;
+    uint32_t Pid;
+};
+```
+
+```
+runtime <- client : MessageHeader { CollectEventPipeTracing }
+    error? -> no response
+runtime -> client : session ID 
+runtime -> client : event stream
+
+...
+
+runtime <- client : stop command
+runtime -> client : session id and stops
+```
+
+### Proposed
+
+Connection:
+```
+runtime <- client : handshake (set length byte array that represents version)
+runtime -> client : sends handshake back as ACK
+runtime <- client : packet containing start code
+runtime -> client : ACK + potential payload
+repeat till closed
+```
+
+### Transport considerations
+
+* All data will be sent in the endianess of the machine
```

Comment:
It would be nice if the specification does not depend on the machine specifics.
If the connection is proxied (to the outside of a virtual machine, for example), then the concept `the machine` is not really a well defined concept.

Practically all plaforms we cared are litte endian. Might as well just fix on that